app/javascript/controllers/disable_controller.js

```diff
@@ -0,0 +1,39 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  connect() {
+    this.element.dataset.action = `${this.element.dataset.action || ""} submit->disable#disableForm`
+    this.uploadCount = 0
+  }
+
+  disableForm() {
+    this.submitButtons().forEach((el) => {
+      el.disabled = true
+      el.value = el.dataset.disableWith || el.value
+    })
+  }
+
+  uploading(disable = true) {
```

Comment:
ah i didn't understand `uploadCount` at first but i think this makes it clear. when an upload begins (which i assume has an action that triggers this method), we tick the count up. and because there can be multiple uploads in progress at the same time, we need to keep track so that even if one finishes early, we don't automatically enable all the submit buttons in case there is still another upload in progress.

Reply:
Exactly